github_actions/lib/dependabot/github_actions/file_fetcher.rb

```diff
@@ -7,11 +7,11 @@ module Dependabot
   module GithubActions
     class FileFetcher < Dependabot::FileFetchers::Base
       def self.required_files_in?(filenames)
-        filenames.any? { |f| f == ".github" }
+        filenames.any? { |f| f.match?("^(\.github|action.ya?ml)$") }
```

Comment:
Little bit of a nit, but:

I was convinced that `match` takes a regular expression as an argument, but it turns out that strings are converted to regular expressions. However, I still feel like we could save a few computation cycles and improve clarity a little bit by passing a regex:

```suggestion
        filenames.any? { |f| f.match?(/^(\.github|action.ya?ml)$/) }
```

And while we're at it, could we extract this into a constant? Something like `FILENAME_PATTERN`, it typically goes at the top of the file in ruby so:

```ruby
  module GithubActions
    class FileFetcher < Dependabot::FileFetchers::
      FILENAME_PATTERN = /^(\.github|action.ya?ml)$/.freeze

      # ...
      filenames.any? { |f| f.match?(FILENAME_PATTERN) }
```

Reply:
I took this from how it looked in other file_fetchers, but I can update this according to the suggestion :)

Reply:
Ah, interesting that the other fetchers are done this way, very understandable you carried over that style that's probably what everyone implementing these has done 😄. I'll have a look at those separately because I do think this is slightly preferable and thanks for updating it 🙇